bitmovin/bitmovin-api-sdk-examples
# EN-13960 Add Encoding Templates examples

# What was done

- ~use latest SDK version (remove `segmentCut` usage in keyframes)~
	- extracted into https://github.com/bitmovin/bitmovin-api-sdk-examples/pull/63
- add Encoding Templates example with async request
	- "Standard VOD Workflow" from Dashboard demos
	- ✅ JS/TS
	- ✅ Java
	- ✅ .NET
	- ✅ Go
	- ✅ Python (SDK's internal HTTP client serializes objects with sorting, which breaks Encoding Templates order)
		- needs SDK `1.209.0` fix: https://github.com/bitmovin-engineering/bitmovin-open-api/pull/1488
	- ⚠️ PHP (internal resource references in Encoding Templates of format `$/.../.../resource-name` break with [PHP string sanitization](https://www.php.net/manual/en/filter.examples.sanitization.php))

# How to test

- check out the repository and branch `feature/EN-13960-encoding-templates`
- select a language of your choice by `$ cd <language>`
- create an `example.properties` file in the language folder with your IDs and credentials:
```
BITMOVIN_API_KEY=...
BITMOVIN_TENANT_ORG_ID=...
S3_OUTPUT_BUCKET_NAME=...
S3_OUTPUT_ACCESS_KEY=...
S3_OUTPUT_SECRET_KEY=...
S3_OUTPUT_BASE_PATH=/bitmovin-api-sdk-examples
HTTP_INPUT_HOST=bitmovin-sample-content.s3.eu-west-1.amazonaws.com
HTTP_INPUT_FILE_PATH=/bbb_sunflower_1080p_60fps_normal.mp4
```
- select an example and run it by `$ ./run-example.sh EncodingTemplates` (`encoding_templates` for Python)
